substrate/frame/staking/src/lib.rs

```diff
@@ -740,26 +743,36 @@ pub struct Nominations<T: Config> {
 ///
 /// This is useful where we need to take into account the validator's own stake and total exposure
 /// in consideration, in addition to the individual nominators backing them.
-#[derive(Encode, Decode, RuntimeDebug, TypeInfo, PartialEq, Eq)]
-pub struct PagedExposure<AccountId, Balance: HasCompact + codec::MaxEncodedLen> {
+#[derive(Encode, Decode, RuntimeDebug, TypeInfo, PartialEq, Eq, MaxEncodedLen)]
+pub struct PagedExposure<
+	AccountId,
+	Balance: HasCompact + MaxEncodedLen,
+	MaxExposurePageSize: Get<u32>,
+> {
 	exposure_metadata: PagedExposureMetadata<Balance>,
-	exposure_page: ExposurePage<AccountId, Balance>,
+	exposure_page: ExposurePage<AccountId, Balance, MaxExposurePageSize>,
 }
 
-impl<AccountId, Balance: HasCompact + Copy + AtLeast32BitUnsigned + codec::MaxEncodedLen>
-	PagedExposure<AccountId, Balance>
+impl<
+		AccountId,
+		Balance: HasCompact + Copy + AtLeast32BitUnsigned + MaxEncodedLen,
+		MaxExposurePageSize: Get<u32>,
+	> PagedExposure<AccountId, Balance, MaxExposurePageSize>
 {
-	/// Create a new instance of `PagedExposure` from legacy clipped exposures.
-	pub fn from_clipped(exposure: Exposure<AccountId, Balance>) -> Self {
-		Self {
+	/// Create a new instance of `PagedExposure` from legacy clipped exposures (now removed).
+	pub fn from_clipped(exposure: Exposure<AccountId, Balance>) -> Result<Self, ()> {
+		let old_exposures = exposure.others.len();
+		let others = WeakBoundedVec::try_from(exposure.others).unwrap_or_default();
+		defensive_assert!(old_exposures == others.len(), "Too many exposures for a page");
```

Comment:
This function is not used, we can make it better or remove it.
```suggestion
	pub fn try_from_clipped(exposure: Exposure<AccountId, Balance>) -> Result<Self, ()> {
		let others = WeakBoundedVec::try_from(exposure.others).map_err(|_| ())?;
```

Reply:
Changed in https://github.com/paritytech/polkadot-sdk/pull/6445/commits/3b7bc950aee8ab20f88fda18331e18b4ec884b0a